Sync `Dual-displays implementation on mobile` to upstream (#56)
nimblehq/showcase-flutter-build-web-desktop: README.md

<p align="center">
  <a href="https://nimblehq.co/">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://assets.nimblehq.co/logo/dark/logo-dark-text-320.png">
      <img alt="Nimble logo" src="https://assets.nimblehq.co/logo/light/logo-light-text-320.png">
    </picture>    
  </a>
</p>

> ---

Reviewer: @doannimble you seem missed this review? 👋

> ---

Author: @luongvo Ah yes, my bad. I've removed it 🙏

- Flutter 3.10.5

- A device support multiple displays. If not, we can try to cast by connecting your Android/iOS phone to other displays.
  Or simulating on the Android phone by enabling `Simulate secondary displays` (Settings -> System -> Developer options
  -> DISPLAY)

<p align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/b971e1a3-606f-42ab-a8a8-62e9ea379486" width="400">

>       <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/b971e1a3-606f-42ab-a8a8-62e9ea379486" width="400">

Reviewer: We will merge https://github.com/nimblehq/showcase_flutter_build_ios_android into upstream and delete it after that so please migrate these assets to https://github.com/nimblehq/showcase-flutter-build-web-desktop 🙏

  ACCESS_TOKEN=
  ```

- Required env variables:
  `1Password: Nimble CX - General`

- Run code generator

    - `$ flutter packages pub run build_runner build --delete-conflicting-outputs`

### Run

> - Run the app with the platform:
> 
>     - Web: `$ flutter run -d chrome`
> 
>     - Windows: `$ flutter run -d windows`
> 
>     - Mobile: `$ flutter run`

Reviewer: since we had instructions below to guide how to run the app on the web and desktop, should we optimize this section like this?

```suggestion
#### Mobile: 

    `$ flutter run`
```

Author: I think we can keep this to make it consistent 🤔

Reviewer: @doannimble, hmm, but that looks repetitive with 2 below sections on how to run the app on the web or windows.

Author: @luongvo yes, for the detail, we should make them repetitive. But this looks like the list of commands, hence, creating another region is odd imo 🙏

Reviewer: @doannimble If you say so, not a big deal.

   `$ flutter run -d chrome`

> If you encounter any issues related to `XMLHttpRequest` or the Dio library when running the app
> locally on the web,
> you can refer to this article for a potential
>
fix: [Fixing Flutter web XMLHttpRequest error with Dio library](https://protocoderspoint.com/flutter-web-xmlhttprequest-error-dio-library-web-issue-fixed/?expand_article=1)
.
> It provides additional information and troubleshooting steps specifically for resolving these issues

> > encountered during local development.

Reviewer: please revert to the original change, as its format is broken now

<img width="1021" alt="image" src="https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/16315358/f0fef700-379d-4fec-9971-141fffe70a3e">

   `$ flutter run -d chrome`

> If you encounter any issues related to `XMLHttpRequest` or the Dio library when running the app
> locally on the web,
> you can refer to this article for a potential
>
fix: [Fixing Flutter web XMLHttpRequest error with Dio library](https://protocoderspoint.com/flutter-web-xmlhttprequest-error-dio-library-web-issue-fixed/?expand_article=1)
.
> It provides additional information and troubleshooting steps specifically for resolving these issues
> encountered during local development.

> ### How to use on the web/desktop:

Reviewer: I think we could have a new main section with the same level with `Run` for this and the below content

```suggestion
## Demonstration

### Flutter for Web and desktop
```

      <img alt="Screenshot" src="https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/88996251/21c069f8-cf9e-4311-97b4-09b3fa457b86" width="200">
    </picture>
</p>

3. To display the cart on the second screen, click `Setup Display` button

<p align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/88996251/7db1ab44-4142-429c-9a11-5ad8cb402ef1" width="400">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/88996251/7db1ab44-4142-429c-9a11-5ad8cb402ef1" width="400">
    </picture>
</p>

4. Now we can update the cart, and it will reflect to the second screen accordingly

<p align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/88996251/b602d5dc-2877-40e4-ab99-a925d5bab0fd" width="400">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/88996251/b602d5dc-2877-40e4-ab99-a925d5bab0fd" width="400">
    </picture>
</p>

> 5. Finally, we click `Pay Now` to notify on the second screen shows the instruction payment the text

Reviewer 2: ```suggestion
5. Finally, we click `Pay Now` to notify the second screen to show the instruction payment text
```

https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/28002315/2c6540de-acf9-4955-973b-b52409c3f20b

## Dual-displays integration on mobile

For Android, you can find the APKs for all versions

> in [Releases](https://github.com/nimblehq/showcase_flutter_build_ios_android/releases)

Reviewer: we need to update the repo. it's now here https://github.com/nimblehq/showcase-flutter-build-web-desktop/releases

1. At the start screen, click `Sign in with CX` to log in automatically with the default account

<p align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/c136d02e-a571-4841-975a-a035851e4ebb" width="400">

>       <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/c136d02e-a571-4841-975a-a035851e4ebb" width="400">

Reviewer: and all as below 👇

      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/020393c0-08fe-487f-99ec-925c832a76b8" width="200">
    </picture>
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/4142672d-1b8a-4a2d-939d-5f34afd4e1aa" width="200">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/4142672d-1b8a-4a2d-939d-5f34afd4e1aa" width="200">
    </picture>
</p>

3. To display the cart on the second screen, click `Setup Display` button

<p align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/a1ff5071-b5c4-4c95-b266-0c908d55bd1f" width="400">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/a1ff5071-b5c4-4c95-b266-0c908d55bd1f" width="400">
    </picture>
</p>

> 4. Now we can update the cart, and it will reflect to the second screen accordingly

Reviewer 2: ```suggestion
4. Now we can update the cart, and it will reflect on the second screen accordingly
```

https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/28002315/2c6540de-acf9-4955-973b-b52409c3f20b

> ## How to use on the app

Reviewer: ```suggestion
### Dual-displays integration on mobile
```

      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/8a87d292-5c09-47a7-8100-7d28bfb908ff" width="400">
    </picture>
</p>

5. Finally, we click `Pay Now` to notify on the second screen shows the instruction payment the text

<p align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/8ea502d8-09a4-44ed-92fa-8ca0506e1a9f" width="300">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/8ea502d8-09a4-44ed-92fa-8ca0506e1a9f" width="200">
    </picture>
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/73b8c6ae-12b8-47c9-ac53-716ae20461d3" width="300">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/73b8c6ae-12b8-47c9-ac53-716ae20461d3" width="200">
    </picture>
</p>

> **Kudos! You know it.**

> ## Package used for dual-display
> 
> [Presentation Displays](https://pub.dev/packages/presentation_displays)
> 
> ## Proof of Work
> 
> https://drive.google.com/file/d/19EjUpbiAurwyoNWKE5BLTzOBQSEQECnJ/view

Reviewer: ```suggestion
#### Package used for dual-display

[Presentation Displays](https://pub.dev/packages/presentation_displays)

#### Proof of Work

https://drive.google.com/file/d/19EjUpbiAurwyoNWKE5BLTzOBQSEQECnJ/view
```

https://github.com/nimblehq/showcase-flutter-build-web-desktop/assets/28002315/2c6540de-acf9-4955-973b-b52409c3f20b

> ## Dual-displays integration on mobile

Reviewer: This should be in lower level

```suggestion
### Dual-displays integration on mobile
```

      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/8a87d292-5c09-47a7-8100-7d28bfb908ff" width="400">
    </picture>
</p>

5. Finally, we click `Pay Now` to notify on the second screen shows the instruction payment the text

<p align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/8ea502d8-09a4-44ed-92fa-8ca0506e1a9f" width="300">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/8ea502d8-09a4-44ed-92fa-8ca0506e1a9f" width="200">
    </picture>
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/73b8c6ae-12b8-47c9-ac53-716ae20461d3" width="300">
      <img alt="Screenshot" src="https://github.com/nimblehq/showcase_flutter_build_ios_android/assets/88996251/73b8c6ae-12b8-47c9-ac53-716ae20461d3" width="200">
    </picture>
</p>

> **Kudos! You know it.**

> ### Package used for dual-display
> 
> [Presentation Displays](https://pub.dev/packages/presentation_displays)
> 
> ### Proof of Work

Reviewer: Then

```suggestion
#### Package used for dual-display

[Presentation Displays](https://pub.dev/packages/presentation_displays)

#### Proof of Work
```
